rcl_yaml_param_parser/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(rcl_yaml_param_parser)
+
+find_package(ament_cmake_ros REQUIRED)
+find_package(yaml REQUIRED)
+find_package(rcutils REQUIRED)
+find_package(rcl REQUIRED)
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic")
```

Comment:
Actually this is a C library so we need to set the flags for the C compiler.
A more generic way to do it is:
```
if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
  add_compile_options(-Wall -Wextra -Wpedantic)
```
This way the option is added for both C and C++.

Compiling with this generates warnings:
```
/home/mikael/work/ros2/bouncy_ws/src/ros2/rcl/rcl_yaml_param_parser/src/parser.c: In function ‘rcl_yaml_node_struct_print’:
/home/mikael/work/ros2/bouncy_ws/src/ros2/rcl/rcl_yaml_param_parser/src/parser.c:318:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (int i = 0; i < param_var->bool_array_value->size; i++) {
                               ^
/home/mikael/work/ros2/bouncy_ws/src/ros2/rcl/rcl_yaml_param_parser/src/parser.c:327:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (int i = 0; i < param_var->integer_array_value->size; i++) {
                               ^
/home/mikael/work/ros2/bouncy_ws/src/ros2/rcl/rcl_yaml_param_parser/src/parser.c:335:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (int i = 0; i < param_var->double_array_value->size; i++) {
                               ^
/home/mikael/work/ros2/bouncy_ws/src/ros2/rcl/rcl_yaml_param_parser/src/parser.c:343:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (int i = 0; i < param_var->string_array_value->size; i++) {
                               ^
/home/mikael/work/ros2/bouncy_ws/src/ros2/rcl/rcl_yaml_param_parser/src/parser.c: In function ‘parse_value’:
/home/mikael/work/ros2/bouncy_ws/src/ros2/rcl/rcl_yaml_param_parser/src/parser.c:619:24: warning: pointer targets in initialization differ in signedness [-Wpointer-sign]
   const char * value = event.data.scalar.value;
                        ^
/home/mikael/work/ros2/bouncy_ws/src/ros2/rcl/rcl_yaml_param_parser/src/parser.c: In function ‘parse_key’:
/home/mikael/work/ros2/bouncy_ws/src/ros2/rcl/rcl_yaml_param_parser/src/parser.c:797:24: warning: pointer targets in initialization differ in signedness [-Wpointer-sign]
   const char * value = event.data.scalar.value;
```

Reply:
Ooops! you are right. Will do that

Reply:
Fixed the CmakeLists.txt and the warnings